File: AI_FSM/Assets/Scripte/FSM/Base/Transition/WaitTransition.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class WaitTransition : Transition
{
    [SerializeField]
    private float waitTime = 5;
    bool isDone = false;
    //FTimerHandle waitTimer;
    public override void InitTranstition()
    {
        StartCoroutine(Ti);
    }
    public override bool IsValidTranstition()
    { 
        return isDone; 
    }
    IEnumerator Ti()
    {
        yield return waitTime;
    }
private void Wait()
    {
        isDone = true;
    }
}

File: AI_FSM/Source/AI_FSM/Private/FSM/BasicTransitions/WaitTransition.cpp
// Fill out your copyright notice in the Description page of Project Settings.


#include "FSM/BasicTransitions/WaitTransition.h"

void UWaitTransition::InitTranstition()
{
	waitTime = FMath::FRandRange(minTime, maxTime);
	GetWorld()->GetTimerManager().SetTimer(waitTimer, this, &UWaitTransition::Wait, waitTime, false);
}

bool UWaitTransition::IsValidTranstition()
{
	return isDone;
}

void UWaitTransition::Wait()
{
	isDone = true;
}
